.github/workflows/integration-test.yml

```
@@ -0,0 +1,108 @@
+name: Integration tests
+on:
+  pull_request: {}
+  push:
+    paths-ignore:
+    - '*.md'
+    - '**/*.md'
+    branches:
+    - main
+permissions:
+  contents: read
+env:
+  DOCKER_REGISTRY: ghcr.io/linkerd
+  DOCKER_TAG: latest
+
+jobs:
+
+  docker-build:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+    - name: Docker build
+      uses: ./.github/actions/docker-build
+      with:
+        docker-registry: ${{ env.DOCKER_REGISTRY }}
+        docker-tag: ${{ env.DOCKER_TAG }}
+        docker-target: linux-amd64
+        component: failover
+    - name: Create artifact with image
+      env:
+        ARCHIVES: /home/runner/archives
+      run: |
+        mkdir -p $ARCHIVES
+        docker save "${{ env.DOCKER_REGISTRY}}/failover:${{ env.DOCKER_TAG }}" > $ARCHIVES/failover.tar
+    # `with.path` values do not support environment variables yet, so an
+    # absolute path is used here.
+    # https://github.com/actions/upload-artifact/issues/8
+    - name: Upload artifact
+      uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2
+      with:
+        name: image-archives
+        path: /home/runner/archives
```

I'm surprised that you can't have actions do the templating
```suggestion
    - name: Upload artifact
      uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2
      with:
        name: image-archives
        path: ${{ env.ARCHIVES }}
```

not a big deal...

---

That was an old limitation. I've just cleaned that up :+1: